webapp/app.py

```diff
@@ -113,28 +127,37 @@ def logout():
     """
     Empty the session, used to logout.
     """
-    flask.session.pop("openid", None)
+    flask.session.pop("authentication_token", None)
 
     return flask.redirect(open_id.get_next_url())
 
 
 @app.route("/login", methods=["GET", "POST"])
 @open_id.loginhandler
 def login_handler():
-    if "openid" in flask.session and "macaroon_root" in flask.session:
+    if "authentication_token" in flask.session:
         return flask.redirect(open_id.get_next_url())
 
-    root = request_macaroon()
-    token = root["metadata"]["token"]
+    response = requests.get(
+        url=f"{ANBOXCLOUD_API_BASE}1.0/token", params=[("provider", "usso")]
```

Comment:
I believe you could use a dictionary for `params` here:

```suggestion
        url=f"{ANBOXCLOUD_API_BASE}1.0/token", params={"provider": "usso"}]
```

A much more friendly format.